rest/api/retrievals-api/push.go

```diff
@@ -0,0 +1,72 @@
+package reportingapi
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/application-research/estuary-metrics/rest/api"
+	"github.com/filecoin-project/lassie/pkg/eventpublisher"
+	"github.com/gin-gonic/gin"
+	"github.com/gofrs/uuid"
+	"github.com/julienschmidt/httprouter"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+func ConfigureRetrievalsPushRouter(router gin.IRoutes) {
+	router.POST("/retrievals/push", api.ConvertHttpRouterToGin(AddRetrievalEvent))
+}
+
+// AddRetrievalEvent publishes a retrieval event into the database
+// @Summary Add Retrival Event
+// @Description Get device usages
+// @Tags Environment
+// @Accept  json
+// @Body multiEventReport body multiEventReport true "multiEventReport"
+// @Success 200 {object}
+// @Failure 500 {object} api.HTTPError
+// @Router /reporting/retrievals/log [post]
+func AddRetrievalEvent(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	var multiRetrEvent multiEventReport
+	resBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return
+	}
+	json.Unmarshal(resBody, &multiRetrEvent)
+
+	// @jcace TODO
```

Comment:
This is going to be on another PR right?

Reply:
Yes that will be over here https://github.com/application-research/estuary-metrics/pull/20